doc/classes/ShapeCast2D.xml

```diff
@@ -42,27 +42,27 @@
 		<method name="get_closest_collision_safe_fraction" qualifiers="const">
 			<return type="float" />
 			<description>
-				The fraction of the motion (between 0 and 1) of how far the shape can move without triggering a collision. The motion is determined by [member target_position].
+				The fraction from the [ShapeCast2D]'s origin to its [member target_position] (between 0 and 1) of how far the shape can move without triggering a collision.
 			</description>
 		</method>
 		<method name="get_closest_collision_unsafe_fraction" qualifiers="const">
 			<return type="float" />
 			<description>
-				The fraction of the motion (between 0 and 1) when the shape triggers a collision. The motion is determined by [member target_position].
+				The fraction from the [ShapeCast2D]'s origin to its [member target_position] (between 0 and 1) of how far the shape must move before triggering a collision.
```

Comment:
Just a suggestion, feel free to disagree:
```suggestion
				The fraction from the [ShapeCast2D]'s origin to its [member target_position] (between 0 and 1) of how far the shape must move to triggering a collision.
```
I'm mostly thinking about users not too familiar with English, "before" could be interpreted as "the collision will NOT be triggered".

Reply:
I agree, that seems like a good change.